wordpress/wp-content/plugins/memberful-wp/src/search_filter.php

```diff
@@ -3,19 +3,30 @@
 add_action( 'pre_get_posts', 'memberful_wp_protect_search' );
 
 function memberful_wp_protect_search( $query ) {
-  // Do not filter search results for admins
+  // Do not protect admin pages.
+  if ( is_admin() ) {
+    return;
+  }
```

Comment:
Similar to `current_user_can()`, can this fail because WP hasn't loaded yet? Should we do this after checking `is_search()`?

Reply:
It can’t. This function is defined in `wordpress/wp-includes/load.php` (one of the first required files), while `wp_get_current_user()` (which is called by `current_user_can()`) is defined in `wordpress/wp-includes/pluggable.php`, which is loaded later.

Also, the function name is misleading. `is_admin()` doesn’t check user capabilities; it checks whether the user is visiting a page in /wp-admin, so it doesn't need to access the current user.

Reply:
Great, thanks!